File: implementations/java/summation/Summation_small.java
import java.io.*;

public class Summation_Small {
    static final int N = 100;
    static long[] arr = new long[N];

    static void setup() {
        for (int i = 0; i < N; i++) {
            arr[i] = (long)(i + 1);
        }
    }

    static long summation() {
        long sum = 0;
        for (int i = 0; i < N; i++) {
            sum += arr[i];
        }
        return sum;
    }

    public static void main(String[] args) throws IOException {
        setup();

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.println("ready");
        System.out.flush();

        String line;
        while ((line = br.readLine()) != null) {
            System.out.println(summation());
            System.out.flush();
        }
    }
}

File: implementations/rust/summation/small/summation_small.rs
use std::io::{self, BufRead, Write};

const N: usize = 100;

static mut ARR: [i64; N] = [0; N];

fn setup() {
    unsafe {
        for i in 0..N {
            ARR[i] = (i + 1) as i64;
        }
    }
}

fn summation() -> i64 {
    unsafe {
        let mut sum: i64 = 0;
        for i in 0..N {
            sum += ARR[i];
        }
        sum
    }
}

fn main() {
    setup();

    println!("ready");
    io::stdout().flush().unwrap();

    let stdin = io::stdin();
    for _ in stdin.lock().lines() {
        println!("{}", summation());
        io::stdout().flush().unwrap();
    }
}